### ISSUE TITLE
gas counter overflow during injection

### ISSUE THREAD

**Issue by grishasobol** [gear-tech/gear#1353]
### Problem

We can make `.wasm` file, which has size less then 512 kByte and it's absolutely correct and suit for our restrictions. But this file cannot be upload because `GasInjection` failed.
You can check this `.wasm`:
[Archive.zip](https://github.com/gear-tech/gear/files/9397166/Archive.zip)

If you try to upload this program then you would receive error like this:
```
2022-08-22 22:15:42.293 DEBUG main pallet_gear::pallet: Code failed to load: GasInjection
``` 
This problem occurs because of `u32` counter overflow:
https://github.com/paritytech/wasm-instrument/blob/1ade161da4e8edc91739a5777b394007d9a48704/src/gas_metering/mod.rs#L426-L429

### Steps

- use gear-idea (or some other tool) to upload attached above wasm file.

**Comment by SabaunT** [comment]
This can be fixed here: https://github.com/paritytech/wasm-instrument/pull/29

**Comment by breathx** [comment]
Fixed in #1324
